dashboard/src/components/EncryptionKey.js

```diff
@@ -103,56 +53,67 @@ const EncryptionKey = ({ isMain }) => {
-      const mutTreatments = await recryptPersonsRelatedDocuments(treatments, previousKey, hashedOrgEncryptionKey);
-      const mutConsultations = await recryptPersonsRelatedDocuments(consultations, previousKey, hashedOrgEncryptionKey);
-      const mutMedicalFiles = await recryptPersonsRelatedDocuments(medicalFiles, previousKey, hashedOrgEncryptionKey);
-
-      const encryptedPersons = await Promise.all(
-        mutPersons.map((person) => preparePersonForEncryption(person, { checkRequiredFields: false })).map(encryptItem)
-      );
-      const encryptedGroups = await Promise.all(
-        groups.map((group) => prepareGroupForEncryption(group, { checkRequiredFields: false })).map(encryptItem)
-      );
+      async function recrypt(path, callback = null) {
+        setEncryptingStatus(`Chiffrement des données... (${path.replace('/', '')}s)`);
+        const cryptedItems = await API.get({
+          skipDecrypt: true,
+          path,
+          query: {
+            organisation: organisation._id,
+            limit: String(Number.MAX_SAFE_INTEGER),
+            page: String(0),
+            after: String(0),
+            withDeleted: true,
+          },
+        });
+        const encryptedItems = [];
+        for (const item of cryptedItems.data) {
+          const recrypted = await decryptAndEncryptItem(item, previousKey, callback);
+          if (recrypted) encryptedItems.push(recrypted);
+        }
+        return encryptedItems;
+      }
 
-      const encryptedActions = await Promise.all(
-        actions.map((action) => prepareActionForEncryption(action, { checkRequiredFields: false })).map(encryptItem)
-      );
-      const encryptedConsultations = await Promise.all(
-        mutConsultations.map(prepareConsultationForEncryption(organisation.consultations, { checkRequiredFields: false })).map(encryptItem)
-      );
-      const encryptedTreatments = await Promise.all(
-        mutTreatments.map((treatment) => prepareTreatmentForEncryption(treatment, { checkRequiredFields: false })).map(encryptItem)
-      );
-      const encryptedMedicalFiles = await Promise.all(
-        mutMedicalFiles
-          .map((medicalFile) => prepareMedicalFileForEncryption(customFieldsMedicalFile)(medicalFile, { checkRequiredFields: false }))
-          .map(encryptItem)
-      );
-      const encryptedComments = await Promise.all(
-        comments.map((comment) => prepareCommentForEncryption(comment, { checkRequiredFields: false })).map(encryptItem)
+      const encryptedPersons = await recrypt('/person', async (data, item) =>
+        recryptPersonRelatedDocuments(data, item._id, previousKey, hashedOrgEncryptionKey)
       );
-      const encryptedPassages = await Promise.all(
-        passages.map((passage) => preparePassageForEncryption(passage, { checkRequiredFields: false })).map(encryptItem)
+      const encryptedConsultations = await recrypt('/consultation', async (data, item) =>
```

Comment:
j'ai mis 3-4 minutes à comprendre ce qu'était `data`
```suggestion
      const encryptedConsultations = await recrypt('/consultation', async ({ documents }, item) =>
```

Reply:
C'est mieux ?